From: Frediano Ziglio <frediano.ziglio@cloud.com>
Subject: x86/boot: Move trampoline to data section

Trampoline is changed and copied to lower memory so to satisfy
UEFI CA memory mitigation W^X requirement.
Although mainly code the code is never executed in the original
location.

Signed-off-by: Frediano Ziglio <frediano.ziglio@cloud.com>
diff --git a/xen/arch/x86/boot/head.S b/xen/arch/x86/boot/head.S
index d867b015d961..038403890230 100644
--- a/xen/arch/x86/boot/head.S
+++ b/xen/arch/x86/boot/head.S
@@ -881,8 +881,9 @@ cmdline_parse_early:
 reloc:
         .incbin "reloc.bin"
 
+#include "x86_64.S"
+
+	.section .init.data, "aw", @progbits
 ENTRY(trampoline_start)
 #include "trampoline.S"
 ENTRY(trampoline_end)
-
-#include "x86_64.S"
